### PR TITLE
feat: implement colletables for pacman

### PR DESCRIPTION
**Pull request by JosuePrado**
# US Related

[US](https://tree.taiga.io/project/joseluis-teran-coffeetime/us/80?milestone=395911 )

## Description

Provide a small description of Pull Request or answer the following questions if it is necessary.

- #### What did I do?
- Add the necessary classes for the collectables 
- #### How did I do it?
- I did it using the architecture 
- #### Why did I do it?
- because it was necessary for the game

## Type of Change

Put an `x` in all the boxes that apply:

- [x] ✨ New Feature (non-breaking change which adds functionality)
- [ ] 🛠️ Bug Fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking Change (fix or feature that would cause existing functionality to change)
- [ ] 🧹 Code Refactor
- [ ] ✅ Build Configuration Change
- [ ] 📝 Documentation
- [ ] 🗑️ Chore

## Pull Request Checklist

Put an `x` in all the boxes that apply:

- [ ] My commit messages are detailed.
- [ ] My code follows the code style of this project.
- [ ] No existing features have been broken without good reason.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests pass.

### CODE REVIEW

File: app/src/main/java/com/pac_man/Collectables/Sphere.java [hunk 1]
```diff
@@ -1,17 +1,48 @@
...
+        isConsume = true;
+    }
+
     @Override
-    public void handleCollision(String[] bodies, Nature nature) {
-        // TODO Auto-generated method stub
-        throw new UnsupportedOperationException("Unimplemented method 'handleCollision'");
+    public void handleCollision(String[] bodies, Nature nature) { 
+        for (String bodyP : bodies) {
+            if (bodyP.equals("Pacman") && nature == Nature.WITH) {
```
Comment (victor-villca) [inline app/src/main/java/com/pac_man/Collectables/Sphere.java]: Nature here should be By

File: app/src/main/java/com/pac_man/Collectables/PowerSphere.java [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    }
+
+    public void consumeSphere() {
+        isConsume = true;
+    }
+
+    @Override
+    public void handleCollision(String[] bodies, Nature nature) {
+        for (String bodyP : bodies) {
+            if (bodyP.equals("Pacman") && nature == Nature.WITH) {
```
Comment (victor-villca) [inline app/src/main/java/com/pac_man/Collectables/PowerSphere.java]: Same thing here, nature should be BY

File: app/src/main/java/com/pac_man/Collectables/APowerUp.java [hunk 3]
```diff
@@ -0,0 +1,10 @@
+package com.pac_man.Collectables;
+
+import com.bridge.renderHandler.sprite.Sprite;
+import com.pac_man.Collisions.ICollisionSubscriber;
+import com.pac_man.characters.Pacman.Block;
+
+public abstract class APowerUp implements ICollisionSubscriber {
+    protected Sprite sprite;
+    protected Block block;
```
Comment (victor-villca) [inline app/src/main/java/com/pac_man/Collectables/APowerUp.java]: If i am not wrong the Block class does not exist anymore in the UML, It should be its interface IBlock. Also check if you have to implement that class or import it from the jar.

**Review by victor-villca — CHANGES_REQUESTED** [review]
You did not update the jar library to the latest version of it, please ask the architecture if necessary for your US.

File: app/src/main/java/com/pac_man/Collectables/Sphere.java (hunk 1 above)
Comment (JosuePrado) [reply app/src/main/java/com/pac_man/Collectables/Sphere.java]: Yeah, you right, it's fixed

File: app/src/main/java/com/pac_man/Collectables/PowerSphere.java (hunk 2 above)
Comment (JosuePrado) [reply app/src/main/java/com/pac_man/Collectables/PowerSphere.java]: Yeah, you right, it's fixed

File: app/src/main/java/com/pac_man/Collectables/APowerUp.java (hunk 3 above)
Comment (JosuePrado) [reply app/src/main/java/com/pac_man/Collectables/APowerUp.java]: I have changed that, Could you please review it again ?

**Review by victor-villca — APPROVED** [review]
NIce fixes.

**Pull request merged**